src/steps/os/linux.rs

```diff
@@ -234,6 +237,42 @@ fn upgrade_redhat(ctx: &ExecutionContext) -> Result<()> {
     Ok(())
 }
 
+fn upgrade_nobara(ctx: &ExecutionContext) -> Result<()> {
+    let sudo = require_option(ctx.sudo().as_ref(), REQUIRE_SUDO.to_string())?;
+    let pkg_manager = which("dnf").unwrap_or_else(|| Path::new("yum").to_path_buf());
+
+    let mut update_command = ctx.run_type().execute(sudo);
+    update_command.arg(&pkg_manager).arg("update");
+    // See https://nobaraproject.org/docs/upgrade-troubleshooting/how-do-i-update-the-system/
+    update_command.args([
+        "rpmfusion-nonfree-release",
+        "rpmfusion-free-release",
+        "fedora-repos",
+        "nobara-repos",
+    ]);
+    update_command.arg("--refresh").status_checked()?;
+
+    let mut upgrade_command = ctx.run_type().execute(sudo);
+    upgrade_command
+        .arg(&pkg_manager)
+        .arg(if ctx.config().redhat_distro_sync() {
+            "distro-sync"
+        } else {
+            "upgrade"
+        });
```

Comment:
If the Nobara suggests that we should use:

```sh
$ sudo dnf distro-sync
```
it might be better to just follow it and use the `distro-sync` option